extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleNativeImage.java

```diff
@@ -26,4 +28,10 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, "com.sun.jndi.ldap.LdapCtxFactory"));
     }
 
+    @BuildStep
+    void runtimeInitialization(BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitializedClass) {
+        runtimeInitializedClass
+                .produce(new RuntimeInitializedClassBuildItem("oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource"));
```

Comment:
Maybe I made a mistake in my original issue description but the class that needs to be registered by reflection is: `oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser`

Reply:
Pushed an update. Can you test it please?

Reply:
These changes are working!

Reply:
Thanks for checking!